src/System.Management.Automation/namespaces/FileSystemProvider.cs

```diff
@@ -1350,9 +1350,10 @@ protected override void InvokeDefaultAction(string path)
                     invokeProcess.Start();
                 }
 #elif CORECLR
-                catch (Win32Exception ex) when (ex.NativeErrorCode == 193)
+                catch (Win32Exception ex) when (ex.NativeErrorCode == 193 || ex.NativeErrorCode == 5)
```

Comment:
Opened issue in corefx https://github.com/dotnet/corefx/issues/22299

For now, we can special case directories as the existing Unix code path will work using `xdg-open` but it currently doesn't fall through to that code